DigitalLearningSolutions.Data/Services/DelegateUploadFileService.cs

```diff
@@ -0,0 +1,290 @@
+            {
+                delegateRow.Error = BulkUploadResult.ErrorReason.EmailAddressInUse;
+                return;
+            }
+
+            if (!RecordNeedsUpdating(delegateUser, delegateRow))
+            {
+                delegateRow.RowStatus = RowStatus.Skipped;
+                return;
+            }
+
+            UpdateDelegate(delegateRow, delegateUser);
+        }
+
+        private void UpdateDelegate(DelegateTableRow delegateRow, DelegateUser delegateUser)
+        {
+            userDataService.UpdateDelegate(
+                delegateUser.Id,
+                delegateRow.FirstName!,
+                delegateRow.LastName!,
+                delegateRow.JobGroupId!.Value,
+                delegateRow.Active!.Value,
+                delegateRow.Answer1,
+                delegateRow.Answer2,
+                delegateRow.Answer3,
+                delegateRow.Answer4,
+                delegateRow.Answer5,
+                delegateRow.Answer6,
+                delegateRow.AliasId,
+                delegateRow.Email!
+            );
+            delegateRow.RowStatus = RowStatus.Updated;
+        }
+
+        private bool RecordNeedsUpdating(DelegateUser delegateUser, DelegateTableRow delegateRow)
+        {
+            if (delegateRow.CandidateNumber != null && (delegateUser.AliasId ?? string.Empty) != delegateRow.AliasId)
+            {
+                return true;
+            }
+
+            if ((delegateUser.FirstName ?? string.Empty) != delegateRow.FirstName)
+            {
+                return true;
+            }
+
+            if (delegateUser.LastName != delegateRow.LastName)
+            {
+                return true;
+            }
+
+            if (delegateUser.JobGroupId != delegateRow.JobGroupId!.Value)
+            {
+                return true;
+            }
+
+            if (delegateUser.Active != delegateRow.Active!.Value)
+            {
+                return true;
+            }
+
+            if ((delegateUser.Answer1 ?? string.Empty) != delegateRow.Answer1)
+            {
+                return true;
+            }
+
+            if ((delegateUser.Answer2 ?? string.Empty) != delegateRow.Answer2)
+            {
+                return true;
+            }
+
+            if ((delegateUser.Answer3 ?? string.Empty) != delegateRow.Answer3)
+            {
+                return true;
+            }
+
+            if ((delegateUser.Answer4 ?? string.Empty) != delegateRow.Answer4)
+            {
+                return true;
+            }
+
+            if ((delegateUser.Answer5 ?? string.Empty) != delegateRow.Answer5)
+            {
+                return true;
+            }
+
+            if ((delegateUser.Answer6 ?? string.Empty) != delegateRow.Answer6)
+            {
+                return true;
+            }
+
+            if ((delegateUser.EmailAddress ?? string.Empty) != delegateRow.Email)
+            {
+                return true;
+            }
+
+            return false;

```

Comment:
Could this just be `return (delegateUser.EmailAddress ?? string.Empty) != delegateRow.Email;`